includes/settings.php

```diff
@@ -454,3 +456,91 @@ function sanitize_settings( $settings ) {
 
 	return $new_settings;
 }
+
+/**
+ * Enable auto-update of plugin.
+ *
+ * @param stdClass $transient Transient cache object.
+ *
+ * @return stdClass
+ */
+function auto_update_plugins_setting( $transient ) {
+	if ( ! DT_IS_NETWORK ) {
+		$valid_license = \Distributor\Utils\get_settings()['valid_license'];
+	} else {
+		$valid_license = \Distributor\Utils\get_network_settings()['valid_license'];
+	}
+
+	if ( ! $valid_license ) {
+		return $transient;
+	}
+
+	$current_version = defined( 'DT_VERSION' ) ? DT_VERSION : null;
+	$update = check_for_updates( $current_version );
+	if ( $update ) {
+		$transient->response['distributor/distributor.php'] = $update;
+	} else {
+		$item = (object) array(
+			'id'            => 'distributor/distributor.php',
+			'slug'          => 'distributor',
+			'plugin'        => 'distributor/distributor.php',
+			'new_version'   => $current_version,
+			'url'           => 'https://github.com/10up/distributor',
+			'package'       => '',
+			'icons'         => array(),
+			'banners'       => array(),
+			'banners_rtl'   => array(),
+			'tested'        => '',
+			'requires_php'  => '',
+			'compatibility' => new \stdClass(),
+		);
+
+		$transient->no_update['distributor/distributor.php'] = $item;
+	}
+
+	return $transient;
+}
+
+/**
+ * Compare latest release version from plugin with the current installed version.
+ *
+ * @param string $current_version Current installed version.
+ *
+ * @return false|object
+ */
+function check_for_updates( $current_version ) {
+	$request = wp_safe_remote_get( 'https://api.github.com/repos/10up/distributor/releases?per_page=1' );
```

Comment:
In other places we make remote requests, we make if we are on VIP first in order to use the right function. We probably want to do that here as well. So something like:
```
if ( function_exists( 'vip_safe_wp_remote_get' ) && Utils\is_vip_com() ) {
    vip_safe_wp_remote_get(...);
} else {
    wp_safe_remote_get(...);
}